instrumentation/openfeign-9.0/javaagent/src/main/java/io/opentelemetry/javaagent/instrumentation/openfeign/OpenFeignSpanNameExtractor.java

```diff
@@ -0,0 +1,15 @@
+/*
+ * Copyright The OpenTelemetry Authors
+ * SPDX-License-Identifier: Apache-2.0
+ */
+
+package io.opentelemetry.javaagent.instrumentation.openfeign;
+
+import io.opentelemetry.instrumentation.api.instrumenter.SpanNameExtractor;
+
+public class OpenFeignSpanNameExtractor implements SpanNameExtractor<ExecuteAndDecodeRequest> {
+  @Override
+  public String extract(ExecuteAndDecodeRequest request) {
+    return request.getRequestTemplate().method() + " " + request.getTarget().name();
```

Comment:
> See https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/trace/semantic_conventions/http.md#name Does this name have a low cardinality?


I have seen the implementation class 'HardCodedTarget' of the interface `Target`, it allows to initialize the instance without the "name" field, but this does not mean that the name field of the instance is null, it will be set to the same as url  that declared in "Proxy Target", this declaration usually occurs at compile time.
https://github.com/OpenFeign/feign/blob/015ce2f3a966631feee4daccb3071ff4342e6ecd/core/src/main/java/feign/Target.java#L61
```java
    public HardCodedTarget(Class<T> type, String url) {
      this(type, url, url);
    }
```

So it can be determined that this is a predictable and enumerable cardinality.But it is impossible to prove that all implementation classes of the interface `Target` comply with it, can we add the type assertion "instance of HardCodedTarget"?